compiler/rustc_hir_typeck/src/cast.rs

```diff
@@ -1080,4 +1086,39 @@ impl<'a, 'tcx> CastCheck<'tcx> {
             },
         );
     }
+
+    /// Attempt to suggest using `.is_empty` when trying to cast from a
+    /// collection type to a boolean.
+    fn try_suggest_collection_to_bool(
+        &self,
+        fcx: &FnCtxt<'a, 'tcx>,
+        err: &mut DiagnosticBuilder<'a, ErrorGuaranteed>,
+    ) {
+        if self.cast_ty.is_bool() {
+            // `peel_refs` is important because slices are always behind a ref
+            let suggest = match self.expr_ty.peel_refs().kind() {
+                ty::Adt(adt, _) if fcx.tcx.lang_items().string() == Some(adt.did()) => true,
+
+                ty::Str | ty::Slice(_) => true,
+                _ => false,
+            };
+
+            // We have to have this `suggest` bool because we cannot do
+            // an `if let` with the lang item check in it.
+            if suggest {
+                // If we can't convert the expression to a string, then try a backup
+                match fcx.tcx.sess.source_map().span_to_snippet(self.expr_span) {
```

Comment:
Use `self.cast_span` (or something similar) to suggest changing *just* the `as bool` part to `.is_empty()` instead of `span_to_snippet` and formatting. It's much easier to understand the change that needs to be made and interacts better with macros.